### DISCUSSION TITLE
can llama do other task except text-generate,like translate

### DISCUSSION THREAD

**Discussion by xyx361100238** [ggml-org/llama.cpp#761] [Q&A]
thanks for your work, it's very helpful!
can llama do other jod:
`./main -m ./models/7B/ggml-model-q4_0.bin -p "Translate English to Frence: it's a nic day!" -n 512`

**Comment by rdhox** [comment]
If I'm correct the 7B model is limited with languages (if I ask if it is able to answer me in french, he will say that yes he can but never does). Try the larger models, they seems to be able to handle different languages.

**Reply by xyx361100238** [reply]
yes， you are right， I tried in 13B model，it is works！

**Comment by KASR** [comment]
You can find a list of the languages on which the model was trained here --> [training-dataset](https://github.com/facebookresearch/llama/blob/main/MODEL_CARD.md#training-dataset)

You should try it in interactive mode, it does appear to work also with the 7B model. However, don't set the temperature too high otherwise it hallucinates quite a bit. The translations appear to be correct (at first glance 😃).

![image](https://user-images.githubusercontent.com/22029863/229793930-a57dafed-7942-4eb0-a3ed-f2282b61ce65.png)

**Comment by xyx361100238** [comment]
Thanks， I'll try temperature
